tools/compare_build_systems.py

```diff
@@ -210,10 +221,72 @@ def CompareOptions(bazel_pattern, bazel_files, cmake_pattern, cmake_files):
     both.update(bazel_options)
     both.update(cmake_options)
     for k in both.keys():
-        if not OptionsAreEqual(bazel_options.get(k, None), cmake_options.get(k, None)):
+        if not OptionsAreEqual(
+            bazel_options.get(k, None),
+            cmake_options.get(k, None),
+            warnings_as_errors,
+        ):
             are_equal = False
     return are_equal
 
+def CompareExternalDependencyVersions():
+    pattern = re.compile(BAZEL_MODULE_REGEX)
+    all_okay = True
+    with open(Path(SDK_ROOT) / "MODULE.bazel", "r") as bazel_module_file:
+        for line in bazel_module_file:
+            maybe_match = pattern.match(line)
+            if not maybe_match:
+                continue
+
+            current_submodule_pin = subprocess.run(
+                ("git", "-C", SDK_ROOT, "rev-parse", f'HEAD:lib/{maybe_match.group("dependency")}'),
+                text=True,
```

Comment:
I guess this is assuming that our submodules are always in the `lib/` subdirectory? Whilst this will _probably_ always be true for `pico-sdk`, I guess moving the `lib/` part of the path into the `MODULE.bazel` file might give you an extra bit of flexibility? :shrug:

Reply:
Sent out https://github.com/raspberrypi/pico-sdk/pull/2040 for this one